### PR TITLE
Use NailGun version 0.15.0

### PR DESCRIPTION
**Pull request by elyezer**
* Use specialized compute resource entity classes.
* Update test_multiple_paths.

Test results:

```python
$ py.test tests/foreman/api/test_multiple_paths.py -kComputeResource
================================== test session starts ===================================
platform darwin -- Python 2.7.6 -- py-1.4.26 -- pytest-2.6.4
collected 295 items

tests/foreman/api/test_multiple_paths.py ......sssss..............

====================== 270 tests deselected by '-kComputeResource' =======================
================= 20 passed, 5 skipped, 270 deselected in 108.98 seconds =================

$ py.test tests/foreman/api/test_docker.py
================================== test session starts ===================================
platform darwin -- Python 2.7.6 -- py-1.4.26 -- pytest-2.6.4
collected 110 items

tests/foreman/api/test_docker.py .............................................................s.sssssss.ssssss..............sssssssssssssssssss

======================== 77 passed, 33 skipped in 1852.73 seconds ========================

$ py.test tests/foreman/api/test_organization.py -ktest_bugzilla_1103157
================================== test session starts ===================================
platform darwin -- Python 2.7.6 -- py-1.4.26 -- pytest-2.6.4
collected 34 items

tests/foreman/api/test_organization.py s

==================== 33 tests deselected by '-ktest_bugzilla_1103157' ====================
======================== 1 skipped, 33 deselected in 7.83 seconds ========================
 
$ py.test tests/foreman/cli/test_host.py -ktest_bz_1177570
================================== test session starts ===================================
platform darwin -- Python 2.7.6 -- py-1.4.26 -- pytest-2.6.4
collected 2 items

tests/foreman/cli/test_host.py s

======================= 1 tests deselected by '-ktest_bz_1177570' ========================
======================== 1 skipped, 1 deselected in 4.96 seconds =========================

$ py.test tests/foreman/ui/test_location.py -ktest_add_computeresource_1
================================== test session starts ===================================
platform darwin -- Python 2.7.6 -- py-1.4.26 -- pytest-2.6.4
collected 159 items

tests/foreman/ui/test_location.py .......

================= 152 tests deselected by '-ktest_add_computeresource_1' =================
======================= 7 passed, 152 deselected in 513.46 seconds =======================
```

UI Organization `test_remove_computeresource_1` failed due to:

```
tests/foreman/ui/test_org.py:691: in test_remove_computeresource_1
    self.assertIsNotNone(element)
E   AssertionError: unexpectedly None
```

Which happens after creating the compute resource.

### CODE REVIEW

**Comment by oshtaier** [comment]
Are we going to have special class per each item from:
choices=(
                    'Docker',
                    'EC2',
                    'GCE',
                    'Libvirt',
                    'Openstack',
                    'Ovirt',
                    'Rackspace',
                    'Vmware',
                )
?

**Comment by oshtaier** [comment]
ACK

**Comment by elyezer** [comment]
@oshtaier yes, that is the final goal. Specialized classes works better than a generic one.

File: robottelo/common/helpers.py [hunk 1]
```diff
@@ -271,10 +271,9 @@ def configure_entities():
       ``robottelo.entity_mixins.Entity`` for more information on the effects of
       this.
     * Set a default value for ``nailgun.entities.GPGKey.content``.
-    * Set the default value for ``nailgun.entities.ComputeResource.url`` if
-      either ``docker.internal_url`` or ``docker.external_url`` is set in the
-      configuration file.
-
+    * Set the default value for ``nailgun.entities.DockerComputeResource.url``
+      if either ``docker.internal_url`` or ``docker.external_url`` is set in
+      the configuration file.
```
Comment (Ichimonji10) [inline robottelo/common/helpers.py]: Nitpick: add a blank line after the end of the list.

File: tests/foreman/api/test_multiple_paths.py [hunk 2]
```diff
@@ -116,14 +117,16 @@ class EntityTestCase(APITestCase):
         entities.Architecture,
         entities.AuthSourceLDAP,
         entities.ComputeProfile,
-        entities.ComputeResource,
+        entities.AbstractComputeResource,
```
Comment (Ichimonji10) [inline tests/foreman/api/test_multiple_paths.py]: I think that `AbstractComputeResource` can be dropped from `tests/foreman/api/test_multiple_paths.py` completely. The `AbstractComputeResource` class doesn't really correspond to anything on the server, and it exists just to make the definitions of `LibvirtComputeResource`, `DockerComputeResource` and so on nicer.

Also, the usage of "abstract" classes is what forced us in to using partials in the first place. If we can drop partials, then this module will become a little less complex and easier to work with.

Comment (elyezer) [inline tests/foreman/api/test_multiple_paths.py]: I will update. I got this doubt when first thought to drop it. But I decided to leave to make sure it also works.

**Comment by Ichimonji10** [comment]
ACK

I'm not sure why that UI test is failing, offhand. The change from a `ComputeResource` to a `LibvirtComputeResource` should not have broken anything.

**Pull request merged**